Singularity/Singularity/Screen/Button.cs

```diff
@@ -0,0 +1,219 @@
+        {
+            if (ButtonHovering != null)
+            {
+                ButtonHovering(this, EventArgs.Empty);
+            }
+        }
+
+        /// <summary>
+        /// Sends out event that button has been right clicked
+        /// </summary>
+        protected virtual void OnButtonClicked()
+        {
+            ButtonClicked?.Invoke(this, EventArgs.Empty);
+        }
+
+
+        /// <summary>
+        /// Draws either ther Texture2D or string representation of
+        /// the button. 
+        /// </summary>
+        /// <param name="spriteBatch"></param>
+        public void Draw(SpriteBatch spriteBatch)
+        {
+            // draw for button that uses a Texture2D
+            if (mIsText == false)
+            {
+                spriteBatch.Draw(mButtonTexture,
+                    mPosition,
+                    null,
+                    mColor,
+                    0f,
+                    new Vector2(0, 0),
+                    mScale,
+                    SpriteEffects.None,
+                    0f);
+
+            }
+
+            // draw for button that uses text
+            else
+            {
+                spriteBatch.DrawString(mFont,
+                    origin: Vector2.Zero,
+                    position: mPosition,
+                    color: mColor,
+                    text: mButtonText,
+                    rotation: 0f,
+                    scale: 1f,
+                    effects: SpriteEffects.None,
+                    layerDepth: 0.2f);
+            }
+        }
+
+
+        /// <summary>
+        /// Updates the button so that it is displayed darker if hovered over,
+        /// as well as sending out button events through user interactions.
+        /// </summary>
+        /// <param name="gametime"></param>
+        public void Update(GameTime gametime)
+        {
+            // if mouse is hovering over button then make draw color gray
+            if (Mouse.GetState().X >= mPosition.X &&
+                Mouse.GetState().X <= mPosition.X + mWidth &&
+                Mouse.GetState().Y > mPosition.Y &&
+                Mouse.GetState().Y <= mPosition.Y + mHeight)
+            {
+                OnButtonHovering();
+                mColor = Color.Gray;
+            }
+
+            // otherwise keep draw color at white
+            else
+            {
+                mColor = Color.White;
+            }
+
+            // check if button has been left clicked
+            if (Mouse.GetState().LeftButton == ButtonState.Pressed &&
+                Mouse.GetState().X >= mPosition.X &&
+                Mouse.GetState().X <= mPosition.X + mWidth &&
+                Mouse.GetState().Y >= mPosition.Y &&
+                Mouse.GetState().Y <= mPosition.Y + mHeight)
+            {
+                OnButtonClicked();
+                mClicked = true;
+            }
+
+
+            // check if left button is also released within button bounds to send out ButtonReleased event
```

Comment:
This is exactly what the InputManager is supposed to be for, right?
(not the in bound, but the notice that it was released)

Reply:
the input manager isnt implemented yet so i cant use it